components/kyma-environment-broker/internal/process/provisioning/get_kubeconfig_test.go

```diff
@@ -11,20 +17,107 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestNewGetKubeconfigStep(t *testing.T) {
-	// given
-	st := storage.NewMemoryStorage()
-	provisionerClient := provisioner.NewFakeClient()
-	step := NewGetKubeconfigStep(st.Operations(), provisionerClient)
-	op := fixture.FixProvisioningOperation("op-id", "inst-id")
-	op.Kubeconfig = ""
-	st.Operations().InsertOperation(op)
-
-	// when
-	newOp, d, err := step.Run(op, logrus.New())
-
-	// then
-	require.NoError(t, err)
-	assert.Zero(t, d)
-	assert.NotEmpty(t, newOp.Kubeconfig)
+const (
+	kubeconfigContents    = "apiVersion: v1"
+	kubeconfigFromRuntime = "kubeconfig-content"
+)
+
+func TestGetKubeconfigStep(t *testing.T) {
+	t.Run("should create k8s client using kubeconfig from RuntimeStatus", func(t *testing.T) {
+		// given
+		st := storage.NewMemoryStorage()
+		provisionerClient := provisioner.NewFakeClient()
+
+		scheme := internal.NewSchemeForTests()
+		err := apiextensionsv1.AddToScheme(scheme)
+
+		k8sCli := fake.NewClientBuilder().WithScheme(scheme).Build()
+
+		step := NewGetKubeconfigStep(st.Operations(), provisionerClient, func(k string) (client.Client, error) { return k8sCli, nil })
+		operation := fixture.FixProvisioningOperation("operation-id", "inst-id")
+		operation.Kubeconfig = ""
+		st.Operations().InsertOperation(operation)
+
+		// when
+		processedOperation, d, err := step.Run(operation, logrus.New())
+
+		// then
+		require.NoError(t, err)
+		assert.Zero(t, d)
+		assert.Equal(t, kubeconfigFromRuntime, processedOperation.Kubeconfig)
+		assert.NotEmpty(t, processedOperation.Kubeconfig)
+		assert.NotEmpty(t, processedOperation.K8sClient)
+	})
+	t.Run("should create k8s client for own_cluster plan using kubeconfig from provisioning parameters", func(t *testing.T) {
+		// given
+		st := storage.NewMemoryStorage()
+		provisionerClient := provisioner.NewFakeClient()
+
+		scheme := internal.NewSchemeForTests()
+		err := apiextensionsv1.AddToScheme(scheme)
+
+		k8sCli := fake.NewClientBuilder().WithScheme(scheme).Build()
+
+		step := NewGetKubeconfigStep(st.Operations(), provisionerClient, func(k string) (client.Client, error) { return k8sCli, nil })
```

Comment:
do not provide provisioner client so we will be sure it won't be called. Assert the `k` is equal to proper kubeconfig content

Reply:
Asserted.